Frontend/business-expense-manager/src/components/dialogs/new-category-budget-dialog copy/new-category-budget-dialog.component.html

```diff
@@ -0,0 +1,20 @@
+<!-- carry on fixing -->
+
+<h2 mat-dialog-title>Add category budget</h2>
+<mat-dialog-content>
+  <form [formGroup]="newCategoryBudgetForm" (ngSubmit)="onSubmit()">
+    <mat-form-field>
+      <mat-label>Category</mat-label>
+        <input matInput formControlName="categoryName" placeholder="Select category">
```

Comment:
This needs to be a select or else they will be able to set budgets for whatever they type in (like potato). The code already exists in cash-flow-dialog: Just update to fit into the format you require. (I used template driven but I think you're using reactive forms)

    <mat-form-field>
      <mat-label>Category</mat-label>
      <mat-select (selectionChange)="checkGoalEligibility($event); checkCategoryCashFlowType(cashFlowForm)" required [(ngModel)]="cashFlow.category" name="category">
        @for (category of categories; track category) {
          <mat-option [value]="category">{{category.name}}</mat-option>
        }
      </mat-select>
      <mat-hint *ngIf="categories?.length === 0">
        No Categories Found.
      </mat-hint>
      <mat-error *ngIf="cashFlowForm.form.controls['category']?.errors">
        <span *ngIf="cashFlowForm.form.controls['category']?.hasError('required')">Required field.</span>
      </mat-error>
    </mat-form-field>